### ISSUE TITLE
ssh: repo urls and arbitrary relative pathes

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1655]
```
    ssh_re = re.compile(r'(?P<proto>ssh)://(?:(?P<user>[^@]+)@)?'
                        r'(?P<host>[^:/#]+)(?::(?P<port>\d+))?'
                        r'(?P<path>[^:]+)(?:::(?P<archive>[^/]+))?$')
```
As the `/` terminating the host[:port] part is mapped into path, a path must begin with a slash and thus is absolute.

The code has a special hack to support `/~` there to specify a path relative to homedir, but not something to specify a path relative to cwd.

**Comment by ThomasWaldmann** [comment]
Also: for what is that `#` that may terminate the host part also?

**Comment by ThomasWaldmann** [comment]
workaround: use a ssh config to give the port, then use the other user@host:path syntax to give relative path.

**Comment by enkore** [comment]
`ssh://user@host:path/./relative/path`

**Comment by ThomasWaldmann** [comment]
1.0-maint?

**Comment by ThomasWaldmann** [comment]
Fixed by #1711.
